### ISSUE TITLE
i18n.init via query string won't set locale correctly

### ISSUE THREAD

**Issue by ianchiu28** [mashpie/i18n-node#391]
Hi,

I have a problem with changing locale via query string in a specific language. (zh-TW)

My configuration like this:
```
i18n.configure({
  // setup some locales - other locales default to en silently
  locales: ['en', 'zh-TW', 'de'],

  // query parameter to switch locale (ie. /home?lang=ch)
  queryParameter: 'lang',

  // where to store json files - defaults to './locales'
  directory: __dirname + '/locales',
});

app.use(i18n.init);
```

And I can see the debug messages say when I run the code:
```
debug will use d:\project\locales\en.json
main.js:43
debug read d:\project\locales\en.json for locale: en
main.js:43
debug will use d:\project\locales\zh-TW.json
main.js:43
debug read d:\project\locales\zh-TW.json for locale: zh-TW
main.js:43
debug will use d:\project\locales\de.json
main.js:43
debug read d:\project\locales\de.json for locale: de
```

Then, I try to use query string to change the language like 
`http://localhost?lang=zh-TW`

And I can see the debug message:
`Overriding locale from query: zh-TW`

But the locale is still 'en'... so I can't change the language

I also try `http://localhost?lang=de` and it worked perfectly

Do I miss anything so that I can't change locale to 'zh-TW' via query string?

In addition, it worked perfectly when I use `accept-language` via http header.
It can change to zh-TW.. so I think the zh-TW.json file is fine.

**Comment by ianchiu28** [comment]
Hi,

After tracing the code, I found that I got different results when using accept-language and query string.
When I use accept-language, I will get locale =  'zh-TW' and lanuguage = 'zh-TW'.
But when I use query string, I will get local = 'en' (default) and language = 'zh-tw'.

**And I found problem can be fixed by setting `preserveLegacyCase` to `false`**

>     // Downcase locale when passed on queryParam; e.g. lang=en-US becomes
>     // en-us.  When set to false, the queryParam value will be used as passed;
>     // e.g. lang=en-US remains en-US.
>     preserveLegacyCase: true


But this option is not metioned on npm page and I can't use this option if I download this module by npm. It's only available on github. The version of npm and github are 0.8.3 and this makes me confused.
